rosbag2_transport/test/rosbag2_transport/test_rosbag2_node.cpp

```diff
@@ -125,6 +135,68 @@ TEST_F(RosBag2NodeFixture, publisher_and_subscriber_work)
   EXPECT_THAT(subscribed_messages[0], StrEq("Hello World"));
 }
 
+TEST_F(RosBag2NodeFixture, generic_subscription_uses_qos)
+{
+  // If the GenericSubscription does not use the provided QoS profile,
+  // its request will be incompatible with the Publisher's offer and no messages will be passed.
+  using namespace std::chrono_literals;
+  std::vector<std::string> test_messages = {"Hello World", "Hello World"};
+  std::string topic_name = "string_topic";
+  std::string type = "test_msgs/Strings";
+  rclcpp::QoS qos = rclcpp::SensorDataQoS();
+  auto publisher = node_->create_publisher<test_msgs::msg::Strings>(topic_name, qos);
+
+  auto subscriber_future_ = std::async(
+    std::launch::async, [this, topic_name, type, qos] {
+      return subscribe_raw_messages(1, topic_name, type, qos);
+    });
+  // Give time to the subscriber to start.
+  std::this_thread::sleep_for(std::chrono::milliseconds(100));
```

Comment:
This test is time dependent, and will be flaky.

Better: switch an atomic bool in the lambda and do some busy wait + a sleep here
Best: condition variable, but I understand it's harder to write